internal/gengapic/gengapic.go

```diff
@@ -348,6 +348,30 @@ func (g *generator) insertRequestHeaders(m *descriptor.MethodDescriptorProto, t
 	}
 }
 
+// insertAutoPopulatedFields generates the conditional initialization of any
+// default-value request fields (for the given method) that are specified for
+// auto-population by autoPopulatedFields.
+//
+// If the field value is not equal to default value at the time of sending the
+// request, implying it was set by the user, or if the field has explicit
+// presence and is set by the user, the field must not be auto-populated by
+// the client. Values automatically populated this way must be reused for
+// retries of the same request.
+func (g *generator) initializeAutoPopulatedFields(servName string, m *descriptor.MethodDescriptorProto) {
+	p := g.printf
+	apfs := g.autoPopulatedFields(servName, m)
+	if len(apfs) == 0 {
+		return
+	}
+	g.imports[pbinfo.ImportSpec{Path: "github.com/google/uuid"}] = true
+	for _, apf := range apfs {
+		apf = snakeToCamel(apf)
+		p("if req != nil && req.%s == nil {", apf)
```

Comment:
You can use these helpers to simplify some of the code here (i.e. remove the `snakeToCamel` call): https://github.com/googleapis/gapic-generator-go/blob/2f1a8d3c867ec551e5a8bd1ce51e181ed4546319/internal/gengapic/gengapic.go#L372-L392

These should really be in `helpers.go` so that they are easier to find.

Reply:
Will refactor to the former, thanks!